static/js/food.js

```diff
@@ -0,0 +1,630 @@
+	var quickpickrec_template = { 
+		  _id: ''
+		, type: 'quickpick'
+		, name: ''
+		, foods: []
+		, carbs: 0
+		, hideafteruse: true
+		, hidden: false
+		, position: HIDDEN
+	};
+
+	var filter = {
+		  category: ''
+		, subcategory: ''
+		, name: ''
+	};
+	
+	var GuiToVal = [
+			// API secret
+			{ "html":"fe_filter_category", 		"type":"dropdownval" , 	"settings":"filter.category" },
+			{ "html":"fe_filter_subcategory", 	"type":"dropdownval" , 	"settings":"filter.subcategory" },
+			{ "html":"fe_filter_name", 			"type":"text" , 		"settings":"filter.name" },
+			
+			{ "html":"fe_id", 					"type":"text" , 		"settings":"foodrec._id" },
+			{ "html":"fe_category", 			"type":"text" , 		"settings":"foodrec.category" },
+			{ "html":"fe_subcategory", 			"type":"text" , 		"settings":"foodrec.subcategory" },
+			{ "html":"fe_name", 				"type":"text" , 		"settings":"foodrec.name" },
+			{ "html":"fe_portion", 				"type":"int" , 			"settings":"foodrec.portion" },
+			{ "html":"fe_unit", 				"type":"dropdownval" , 	"settings":"foodrec.unit" },
+			{ "html":"fe_carbs", 				"type":"int" , 			"settings":"foodrec.carbs" },
+			{ "html":"fe_gi", 					"type":"int" , 			"settings":"foodrec.gi" },
+
+			{ "html":"fe_quickpick_showhidden",	"type":"checkbox" , 	"settings":"showhidden" }
+			
+	];
+	
+	//https://www.iconfinder.com/iconsets/musthave
+	var icon_add = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAABa0lEQVQ4T6WTzysEYRjHP+/Mrv2hHOTuqJRNOfgPSORHokg4OClHcnVzVygHF6WIcuHMnyCHVRyI3ZYxa23vzOzs7LzamaI0e5i89fTWt/f5vPV5n1cQsXLbHepvfLv5JaLORoZNwMbyFo5vYfsWB0c7xAasLa5T/vCg45Oj48P4gJWFVYxCA63L5PzkND5gfm4Jo+Chd5W5OrtsDYgS1pQ1OTuNUfTQO8tcX9xE+QugYnS/X81MzGP7MpTWkEFVZY1KxcVPV3h27zAtA+oCagIcDfWUCgEje31qfHwK06gHjaF5iXQcHCV5lHmqqgQCNEAI0IsavCVDwNBurxoeGwmaAkDDwvYsqtIh//6AJUoklP97s62BbJYeAqIcpJNZsoM+r2aVbKKOekiBL8An3BuAEiGg1SSKAYnttpFxPdR9Jv4zipxFTUuQKqsfYbFGWfTYuO06yRfxIyweoLuG+iMsFuBfvzFy7FqE33vs2BFqlfN5AAAAAElFTkSuQmCC";
+	var icon_remove = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAACrElEQVQ4T42Ty2sTQRzHv5tmk2yyjRNtpfZhL8V6s2KoUNC2XqwgaCsVQcGiFqpHi0c9iRdR/ANE9KR40FIQX4cueKoPaKFoLdSYNtE0abKT1+5s9iW7aUMiHtzTzO7v85md+c6PA4DrHbsPCKIgOWO1pA7dT6YXnXH949SE/F63pqwZtRrO+SCKgjQ5NUV+azpmHj2krMwaJC4c8Erj+/eRyloMMwWFKgbn1nC3ervlK1evkXBLGBZT8SOewotnTylTNLdgeg/pDgZDC2cPHSR8bB22DVC9hFe0SG/H0xFXcHlykjRHRDBWgJcZSCY38Xx2lhqMnRYE34Px/sN9vlQWeoHBAx2yXsRruVAVuFsIBaSJ8+eJGPaBqQV4NROJjTzez89jLBoFn6FgybQL54wS3uTyVDFQ3cL2IYpBv3RhdJSIIQ80tQyv7gEqJvS8AmUlBs7UXPhtjtZgh3UFNYngk86NHCfNAg9dMwHVBPu+CpsVkTXKeJeVG+AGgTOZ3tt6MSKKjy+NjEBjFrR4ElZmA4pdxstMFsyyJu6tZZ7Ux9vwB6EAL50ZGiRECEPPUOixVTRxHlicgSVWxEdZpuZWfNuS2hk48NjwMIkIYZglBnV5Cbqtws/5IaAJmsfCglrEl2y2QeKmEBJ80tixKmxrFpSVr0gV0viQoxho2YUuPohmeFD22PiklLC4ma5JuBvdrfLJI0dJd0s7bM0ES8aR/BXDXGaTskqlL+D3Lwy0tZEePoAd4EA5YF4tYymdonfjmQh3s6dTPjU4SHYGwjAKecSXFyGlM1TdytntE56T+ts7SC/vhw3gm6njc2Kd3vm5Ub1IwQAvnYhGiZpYw1wiWYPrIw7wnBTt7CLOOwdmut14kQQvqt24tfK/utGR6LaF+iRqMf4N/O/8D28HiiCRYqzAAAAAAElFTkSuQmCC";
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769498828/issues/source?bid=2452374&fileBranchId=2459081#l59)